File: infix_to_postfix.py
Operators = set(['+', '-', '*', '/', '(', ')', '^'])  # collection of Operators

Priority = {'+':1, '-':1, '*':2, '/':2, '^':3} # dictionary having priorities of Operators
 
 
def infixToPostfix(expression): 
    stack = [] # initialization of empty stack

    output = []

    

    for character in expression:

        if character not in Operators:  # if an operand append in postfix expression

            output.append(character)

        elif character=='(':  # else Operators push onto stack

            stack.append('(')

        elif character==')':

            while stack and stack[-1]!= '(':

                output.append(stack.pop())

            stack.pop()

        else: 

            while stack and stack[-1]!='(' and Priority[character]<=Priority[stack[-1]]:

                output.append(stack.pop())

            stack.append(character)

    while stack:

        output.append(stack.pop())

    return ' '.join(output)


infix = ['8', '^', '2', '+', '6', '-', '2', '*', '10', '/', '2']
postfix = infixToPostfix(infix)
print('infix: ', ' '.join(infix))
print('postfix: ', postfix)

File: infixToPostfix.js
const operators = ['+', '-', '*', '/', '(', ')', '^']

const priority = { '+': 1, '-': 1, '*': 2, '/': 2, '^': 3 }

function infixToPostfix(expression) {
    let open = 0
    let closed = 0
    const stack = []
    const output = []

    expression.forEach((character) => {
        if (!operators.includes(character)) {
            output.push(character)
        }
        else if (character === '(') {
            stack.push('(')
            open += 1
        }
        else if (character === ')') {
            while (stack.length > 0 && stack[stack.length - 1] !== '(') {
                output.push(stack.pop())
                closed += 1
            }
            stack.pop()
        }
        else {
            while (stack.length > 0 && stack[stack.length - 1] !== '(' && priority[character] <= priority[stack[stack.length - 1]]) {
                output.push(stack.pop())
            }
            stack.push(character)
        }
    })
    while (stack.length > 0) {
        output.push(stack.pop())
    }
    console.log('stack after conversion: ', stack)
    console.log('Parens: open, closed: ', open, closed)
    return output.join(' ')
}

const testInput = ['8', '^', '2', '+', '6', '-', '2', '*', '10', '/', '2']
const postFixExpression = infixToPostfix(testInput)
console.log('testInput: ', testInput.join(' ')) 
console.log('postfix: ', postFixExpression)

const testInput2 = ['1', '+', '(', '(', '2', '*', '8', ')']
const postFixExpression2 = infixToPostfix(testInput2)
console.log('testInput2: ', testInput2.join(' ')) 
console.log('postfix2: ', postFixExpression2)